documents/entityrelationship/repository.go

```diff
@@ -29,14 +31,58 @@ func newDBRepository(db storage.Repository, docRepo documents.Repository) reposi
 	return r
 }
 
-// Find returns a EntityRelationship based on a entity id and a targetDID
-func (r *repo) Find(entityIdentifier []byte, targetDID identity.DID) (*EntityRelationship, error) {
-	// todo not implemented
-	return nil, nil
+// Find returns the latest (second if revoked) version of an EntityRelationship based on a entity id and a targetDID
+// Note that we assume a case of maximum two versions of an EntityRelationship document
+func (r *repo) FindEntityRelationshipIdentifier(entityIdentifier []byte, ownerDID, targetDID identity.DID) ([]byte, error) {
+	relationships, err := r.db.GetAllByPrefix(string(ownerDID[:]))
+	if err != nil {
+		return nil, err
+	}
+
+	if relationships == nil {
+		return nil, documents.ErrDocumentNotFound
+	}
+
+	for _, r := range relationships {
+		e, ok := r.(*EntityRelationship)
+		if !ok {
+			continue
+		}
+		if bytes.Equal(e.EntityIdentifier, entityIdentifier) {
+			if targetDID.Equal(*e.TargetIdentity) {
+				return e.Document.DocumentIdentifier, nil
+			}
+		}
+	}
+	return nil, documents.ErrDocumentNotFound
 }
 
-// EntityExists returns true if a entity relationship exists for a given entity identifier and target did
-func (r *repo) EntityExists(entityIdentifier []byte, targetDID identity.DID) (bool, error) {
-	// todo not implemented
-	return false, nil
+// ListAllRelationships returns a list of all entity relationship identifiers in which a given entity is involved
+func (r *repo) ListAllRelationships(entityIdentifier []byte, ownerDID identity.DID) (map[string][]byte, error) {
+	relationships, err := r.db.GetAllByPrefix(string(ownerDID[:]))
+	if err != nil {
+		return nil, err
+	}
+
+	if relationships == nil {
+		return nil, documents.ErrDocumentNotFound
+	}
+
+	all := make(map[string][]byte)
+	for _, r := range relationships {
+		e, ok := r.(*EntityRelationship)
+		if !ok {
+			continue
+		}
+		_, found := all[string(e.Document.DocumentIdentifier)]
+		if bytes.Equal(e.EntityIdentifier, entityIdentifier) && !found {
+			all[string(e.Document.DocumentIdentifier)] = e.Document.DocumentIdentifier
+		}
+	}
+
+	if len(all) == 0 {
+		return nil, documents.ErrDocumentNotFound
```

Comment:
I do not think this is necessarily and error. I would return an empty map and no error

Reply:
okay, will change, makes sense